node/src/store/utils.rs

```diff
@@ -0,0 +1,548 @@
+use std::fmt::Display;
+use std::iter::once;
+use std::ops::RangeInclusive;
+
+use celestia_types::ExtendedHeader;
+use itertools::Itertools;
+use serde::Serialize;
+use smallvec::{IntoIter, SmallVec};
+use thiserror::Error;
+
+use crate::executor::yield_now;
+use crate::store::{Result, StoreError};
+
+pub(crate) const VALIDATIONS_PER_YIELD: usize = 4;
+
+pub type HeaderRange = RangeInclusive<u64>;
+
+pub(crate) trait RangeLengthExt {
+    fn len(&self) -> u64;
+}
+
+impl RangeLengthExt for RangeInclusive<u64> {
+    fn len(&self) -> u64 {
+        self.end() - self.start() + 1
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Default, Serialize)]
+pub struct HeaderRanges(SmallVec<[RangeInclusive<u64>; 2]>);
+
+#[derive(Debug, Error)]
+pub enum HeaderRangeError {
+    #[error("Overlapping range: {0}, {1}")]
+    RangeOverlap(u64, u64),
+}
+
+impl HeaderRanges {
```

Comment:
Move anything releated to `HeaderRanges`/`HeaderRangesIterator` in a `header_ranges.rs` file.

Reply:
split out the header ranges stuff into `store/header_ranges.rs`